New, static site starting from Gatsby starter kit

shilman opened the pull request:
Issue: #5 

### What I did

Ported the old CRA app to gatsby so that the site would be statically generated and indexable from Google. Gatsby and CRA work a bit differently, so I had to hack around to get things working properly.

Major differences from old site:
- Now, instead of authoring `.js` files for content, we author `.md` files, which is more user-friendly. They also render properly in Github source code tree.
- In old site, URLs were of the form `/docs/cat/section/item`. In the new site, `/docs/cat/section/item/`.
- In the old site, you could navigate to `/docs/cat/section` and it would dynamically pick out the first `item` in the section. In the new site, that is handled in the navigation, so all links navigate to specific articles/items. Linking to `/docs/cat/` or `/docs/cat/section` will 404.
- In the old site there were multiple categories. In the new site, there is a single category, `react-storybook`, for now.
- Removed category navigation at the top of the docs page for now.

<img width="499" alt="introduction_-_react_storybook" src="https://cloud.githubusercontent.com/assets/488689/25606905/3def09e6-2f57-11e7-8428-0e378cdec77a.png">

### Still needed

- [ ] favicon
- [ ] Use react-router `Link` instead of `a`

### How to test

```
npm install
npm run storybook
npm run develop
```

### Deploying

Deployment is handled using the `gh-pages` package, that was used by the Gatsby starter kit. I haven't tested it yet.

shilman committed:
New, static site starting from Gatsby starter kit

Work in progress: ported storybooks content

Still needs:
- two-level nav hierarchy
- styling improvements
- stories ;-)

UsulPro commented:
@shilman It the great start! I like the gatsby! I'd be happy to help with it 👍

shilman commented:
Thanks @UsulPro! Let's coordinate with @mnmtanish who is also going to help. Since @mnmtanish worked on the original CRA app, he may have some strong opinions about the transition. I just wanted to get things going because currently https://storybooks.js.org is not getting indexed by Google.

The biggest missing items from the old site right now are:
- [ ] two-level menus on the left + responsive behavior
- [ ] styling
- [ ] edit on github link
- [ ] storybook 😄

shilman committed:
Add edit-this-page link

shilman committed:
Rename doc.md to doc/index.md

Use index.md so that images show properly in github, e.g.:

```md
![Login Form](../static/login_form.png)
```

shilman committed:
Global styling

shilman committed:
Remove auto-generated files and fix deprecation errors

UsulPro commented:
a few more issues with old site:

- [ ] wrong navigation menu behavior on Chrome Android
- [ ] missing section with full API description
- [ ] missing section with Storybook related publications/media

arunoda commented:
This is pretty cool.

shilman commented:
@UsulPro @mnmtanish finally figured out how to merge the old code with gatsby, so hold off on contributing until I check in my latest iteration.

@UsulPro I think those issues with previous site probably deserve to be standalone issues. Let's work it out as soon as I am back to parity with the original site. 😸

shilman committed:
Use original docs components and styling inside gatsby

- get rid of sections concept temporarily
- markdown is already parsed to HTML by gatsby

shilman commented:
@UsulPro @mnmtanish Got it working with some hackery! Please check it out and let me know wha you think. You can now run storybooks. I think we can merge with a little more work.

shilman commented:
Major differences from old site:
- In old site, URLs were of the form `/docs/cat/section/item`. In the new site, `/docs/cat/section/item/`.
- In the old site, you could navigate to `/docs/cat/section` and it would dynamically pick out the first `item` in the section. In the new site, that is handled in the navigation, so all links navigate to specific articles/items.
- In the old site there were multiple categories. In the new site, there is a single category, `react-storybook`, for now.

shilman committed:
Merge branch 'source' into 5-gatsby-static-site

Resolve conflicts merging source changes

# Conflicts:
#	components/Docs/Container/index.js
#	src/docs/react-storybook/configurations/custom-webpack-config.js

shilman committed:
Restore CNAME

shilman committed:
Cleanup unused function

usulpro reviewed (approved):
Looks great! In general, everything runs fine. Should we merge it and consistently solve issues?

UsulPro commented:
I tried to test deploying by this steps:
```shell
npm run build
gh-pages -r https://github.com/UsulPro/storytest.github.io.git -d public -o origin -b master
```
Everything passed without any errors. Result is https://usulpro.github.io/storytest.github.io/
But at this moment available only the home page

usulpro committed:
Add favicon

shilman commented on html.js:
```diff
@@ -5,6 +5,7 @@ import DocumentTitle from 'react-document-title'
 import { prefixLink } from 'gatsby-helpers'
 import typography from './utils/typography'
 import { colors } from 'utils/colors'
+import favicon from './design/homepage/storybook-icon.png'
```
@UsulPro seems unused?

shilman committed:
Fix favicon

usulpro commented on html.js:
```diff
@@ -28,7 +28,7 @@ class HTML extends Component {
           />
           <title>{title}</title>
-          <link rel="icon" href="favicon.ico" type="image/x-icon" />
+          <link rel="icon" href={favicon} type="image/x-icon" />
```
That is exactly what I mean! Sorry, I made a typo yesterday
funny that it doesn't cause an error in console since you added `favicon.ico` to the project root

shilman replied:
thanks for figuring it out!

shilman committed:
Remove container wrapper to fix layout issues

shilman committed:
manually add CNAME to build process

shilman committed:
Ignore public files generated by gatsbyjs

shilman committed:
Fix code highlighting

shilman commented:
@UsulPro regarding your test deploy, there is a variable `linkPrefix = "/"` in `config.toml` that you can adjust if you are deploying to another location. I'm not sure I respect it properly in the code because I was trying to reuse as much of the previous site as possible.
